packages/core/src/common/workspace-utils.ts

```diff
@@ -0,0 +1,37 @@
+/********************************************************************************
+  * Copyright (C) 2018 Ericsson and others.
+  *
+  * This program and the accompanying materials are made available under the
+  * terms of the Eclipse Public License v. 2.0 which is available at
+  * http://www.eclipse.org/legal/epl-2.0.
+  *
+  * This Source Code may also be made available under the following Secondary
+  * Licenses when the conditions for such availability set forth in the Eclipse
+  * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+  * with the GNU Classpath Exception which is available at
+  * https://www.gnu.org/software/classpath/license.html.
+  *
+  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+  ********************************************************************************/
+
+import { Path } from './path';
+
+export namespace WorkspaceUtils {
+
+    /**
+     * Returns shortened workspace path if applicable
+     * @param workspace
+     * @param home
+     */
+    export function getShortenedPath(workspace: string, home?: string): string {
+        const path = new Path(workspace);
+        const dirPath = path.dir.toString();
```

Comment:
I think this function should not strip away the last portion of the path.  If I call

    getShortenedPath("/home/simark/something", "/home/simark")

I should get back `"~/something"`.